idemp-base/src/org/bandahealth/idempiere/base/modelevent/UserModelEvent.java

```diff
@@ -47,11 +47,26 @@ private void updateBusinessPartner(MUser user) {
 			MBPartner_BH businessPartner = (MBPartner_BH) MTable.get(Env.getCtx(), MBPartner_BH.Table_Name)
 					.getPO(user.getC_BPartner_ID(), user.get_TrxName());
 
-			businessPartner.setBH_Birthday(user.getBirthday());
-			businessPartner.setBH_EMail(user.getEMail());
-			businessPartner.setBH_Phone(user.getPhone());
+			boolean didUpdateBusinessPartner = false;
+			if (user.getBirthday() != null && (businessPartner.getBH_Birthday() == null ||
```

Comment:
With the new logic it's no longer possible to nullify a value? Say a user was erroneously saved with a wrong optional field (e.g phone) and the system user nullifies the field. This means the bp will still retain wrong data? Thanks

Reply:
Yep, this was a miss. I'll add it.

Reply:
I've undone this change, though can't test without the [go-2405](https://issues.bandahealth.org/issue/go-2405/Update-how-value-column-used-on-products) fix. Will see if the problem re-appears afterwards and I'll create a separate issue to address that problem when/if it occurs.